<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>StockFinder</class>
 <widget class="QMainWindow" name="StockFinder">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>545</width>
    <height>352</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>智能选股器</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="grpSearchRange">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>10</y>
      <width>151</width>
      <height>141</height>
     </rect>
    </property>
    <property name="title">
     <string>搜索范围</string>
    </property>
    <widget class="QWidget" name="">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>133</width>
       <height>111</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QCheckBox" name="cbxShanghaiMain">
        <property name="text">
         <string>上海主板（60****）</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="cbxShenZhenMain">
        <property name="text">
         <string>深圳主板（000***）</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="cbxShenZhenSmall">
        <property name="text">
         <string>中小板（002***）</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="cbxShenZhenNew">
        <property name="text">
         <string>创业板（300***）</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="cbxShanghaiScience">
        <property name="text">
         <string>科创板（688***）</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QPushButton" name="btnStartSearch">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>200</y>
      <width>151</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>开始搜索</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnGetAllStockList">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>170</y>
      <width>151</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>获取沪深全部股票列表</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="grpTechnicalIndex">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>351</width>
      <height>141</height>
     </rect>
    </property>
    <property name="title">
     <string>技术指标</string>
    </property>
    <widget class="QWidget" name="">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>327</width>
       <height>108</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QCheckBox" name="cbxConsecutiveTrend">
          <property name="text">
           <string>股价连续</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="rbnConsecutiveUp">
          <property name="text">
           <string>上涨</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="rbnConsecutiveDown">
          <property name="text">
           <string>下跌</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spbConsecutiveUpDays">
          <property name="suffix">
           <string>天</string>
          </property>
          <property name="maximum">
           <number>10</number>
          </property>
          <property name="value">
           <number>3</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QCheckBox" name="cbxAverageVolume">
          <property name="text">
           <string>昨日成交量超过</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spinBox">
          <property name="suffix">
           <string>日平均</string>
          </property>
          <property name="minimum">
           <number>5</number>
          </property>
          <property name="maximum">
           <number>60</number>
          </property>
          <property name="value">
           <number>20</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="doubleSpinBox">
          <property name="suffix">
           <string>倍</string>
          </property>
          <property name="decimals">
           <number>1</number>
          </property>
          <property name="minimum">
           <double>1.000000000000000</double>
          </property>
          <property name="maximum">
           <double>10.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>2.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QCheckBox" name="cbxAveragePriceBias">
          <property name="text">
           <string>昨日股价</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="cbbPriceAverageComparison">
          <property name="editable">
           <bool>false</bool>
          </property>
          <property name="currentText">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spbPriceDeviationBase">
          <property name="suffix">
           <string>日均线</string>
          </property>
          <property name="minimum">
           <number>5</number>
          </property>
          <property name="maximum">
           <number>60</number>
          </property>
          <property name="value">
           <number>20</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="spbPriceDeviation">
          <property name="suffix">
           <string>%</string>
          </property>
          <property name="decimals">
           <number>1</number>
          </property>
          <property name="minimum">
           <double>1.000000000000000</double>
          </property>
          <property name="maximum">
           <double>10.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>2.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="grpCompanyQuality">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>160</y>
      <width>351</width>
      <height>141</height>
     </rect>
    </property>
    <property name="title">
     <string>基本面指标</string>
    </property>
    <widget class="QWidget" name="">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>331</width>
       <height>108</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QCheckBox" name="cbxPriceEarning">
          <property name="text">
           <string>市盈率最低</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spbPriceEarningMin">
          <property name="suffix">
           <string>倍</string>
          </property>
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="value">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblPriceEarningMax">
          <property name="text">
           <string>最高</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spbPriceEarningMax">
          <property name="suffix">
           <string>倍</string>
          </property>
          <property name="minimum">
           <number>5</number>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="value">
           <number>50</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QCheckBox" name="cbxPriceBook">
          <property name="text">
           <string>市净率最低</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spbPriceBookMin">
          <property name="suffix">
           <string>倍</string>
          </property>
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>10</number>
          </property>
          <property name="value">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblPriceBookMax">
          <property name="text">
           <string>最高</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spbPriceBookMax">
          <property name="suffix">
           <string>倍</string>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>50</number>
          </property>
          <property name="value">
           <number>5</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <widget class="QCheckBox" name="cbxTotalShare">
          <property name="text">
           <string>总股本最小</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spbTotalShareMin">
          <property name="suffix">
           <string>亿</string>
          </property>
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>10</number>
          </property>
          <property name="value">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblTotalShareMax">
          <property name="text">
           <string>最大</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spbTotalShareMax">
          <property name="suffix">
           <string>亿</string>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>10000</number>
          </property>
          <property name="value">
           <number>10</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QCheckBox" name="cbxTotalAsset">
          <property name="text">
           <string>市值最小</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spbTotalAssetsMin">
          <property name="suffix">
           <string>亿</string>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="value">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblTotalAssetsMax">
          <property name="text">
           <string>最大</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spbTotalAssetsMax">
          <property name="suffix">
           <string>亿</string>
          </property>
          <property name="minimum">
           <number>10</number>
          </property>
          <property name="maximum">
           <number>99999</number>
          </property>
          <property name="value">
           <number>100</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>545</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>btnStartSearch</sender>
   <signal>clicked()</signal>
   <receiver>StockFinder</receiver>
   <slot>search_all_stocks()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>183</x>
     <y>171</y>
    </hint>
    <hint type="destinationlabel">
     <x>83</x>
     <y>355</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnGetAllStockList</sender>
   <signal>clicked()</signal>
   <receiver>StockFinder</receiver>
   <slot>get_all_stocks()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>61</x>
     <y>178</y>
    </hint>
    <hint type="destinationlabel">
     <x>57</x>
     <y>196</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>search_all_stocks()</slot>
  <slot>get_all_stocks()</slot>
 </slots>
</ui>
